fetch-utils.js

```diff
@@ -44,3 +44,8 @@ export async function deleteAllList() {
     const user = getUser();
     await client.from('lists').delete().eq('user_id', user.id);
 }
+
+export async function deleteAllBought() {
+    const user = getUser();
+    await client.from('lists').delete().eq('user_id', user.id).match({ bought: true });
```

Comment:
Since our RLS policy is telling Supabase only to allow users to interact with the data associated with their user_id, adding the logic to check that the user_id column is equal to the user.id is not necessary on the front end. If you have questions about RLS, let me know!